core/arch/arm/kernel/ree_fs_ta_buffered.c

```diff
@@ -0,0 +1,102 @@
+// SPDX-License-Identifier: BSD-2-Clause
```

Comment:
This content of this file you just as well gone into `core/arch/arm/kernel/ree_fs_ta.c` since they are closely linked any way.

Reply:
Not so clean or easy because of the different definitions for `struct user_ta_store_handle`. Also it would add big `#ifdef` sections in the `.c` file, how is that better than a separate file?

Reply:
The different local versions `struct user_ta_store_handle` are a bit confusing so I wouldn't mind casting to different names. However, this and in one or multiple files is a matter of taste, so I leave it to you to decide.

Reply:
Reworked that also. Properly casting is certainly cleaner. As for one or multiple files, I normally prefer several files but this code is quite small and having two files does not prevent one #Ifdef in `ree_fs_ta.c` anyway, so I have done as you suggested.